pipelines/library/src/common/VersionInfo.groovy

```diff
@@ -148,53 +150,66 @@ class VersionInfo {
 
     // Match JDK9+
     private boolean match223(versionString) {
+        // jdk9+ version string format:
+        //   jdk-<major>[.<minor>][.<security>][.<fix>][-<pre>]+<build>[-<opt>]
+        // eg.
+        // jdk-15+36
+        // jdk-11.0.9+11
+        // jdk-11.0.9.1+1
+        // jdk-11.0.9-ea+11
+        // jdk-11.0.9+11-202011050024
+        // jdk-11.0.9+11-adhoc.username-myfolder
+        //
         //Regexes based on those in http://openjdk.java.net/jeps/223
-        // Technically the standard supports an arbitrary number of numbers, we will support 3 for now
-        final vnumRegex = "(?<major>[0-9]+)(\\.(?<minor>[0-9]+))?(\\.(?<security>[0-9]+))?"
-        final preRegex = "(?<pre>[a-zA-Z0-9]+)"
+        // Technically the standard supports an arbitrary number of numbers, we will support 4 for now
+        final vnumRegex = "(?<major>[0-9]+)(\\.(?<minor>[0-9]+))?(\\.(?<security>[0-9]+))?(\\.(?<fix>[0-9]+))?"
+        final preRegex = "(?<pre>[a-zA-Z]+)"
```

Comment:
according to the 223 standard, numbers are allowed in the pre

Reply:
from the jdk11 code:
```
      # Only [a-zA-Z] is allowed in the VERSION_PRE. Outer [ ] to quote m4.
      [ VERSION_PRE=`$ECHO "$with_version_pre" | $TR -c -d '[a-z][A-Z]'` ]
      if test "x$VERSION_PRE" != "x$with_version_pre"; then
        AC_MSG_WARN([--with-version-pre value has been sanitized from '$with_version_pre' to '$VERSION_PRE'])
      fi

```

Reply:
also the version string is always built as:
```
VERSION_STRING=$VERSION_NUMBER${VERSION_PRE:+-$VERSION_PRE}+$VERSION_BUILD${VERSION_OPT:+-$VERSION_OPT}
```
So wondering if things have moved on somewhat since 223?

Reply:
It was updated in http://openjdk.java.net/jeps/322 but that does not really define what PRE means, and implies it is the same as 223